[blackpanther9229] iP

blackpanther9229 opened the pull request:
**Nah 1.1**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))



Nah frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/blackpanther9229/ip/releases/tag/Level-9).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:

- [X]  Managing tasks
- [X]  Managing deadlines
- [X]  Remindering events 

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```

    public static void main(String[] args) {

        Nah nah = new Nah(hardDisk);
        nah.run();

    }

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

blackpanther9229 committed:
Create Nah class with logo, greet() method and exit() method

blackpanther9229 committed:
Create new class with logo, greeting function and exit function

blackpanther9229 committed:
change main function to take input and respond correctly

blackpanther9229 committed:
Nah.java : Change main() method to take input and respond correctly

blackpanther9229 committed:
create a list of task and add function, adjust the main function

blackpanther9229 committed:
Nah.java : Create a list of task and add() method

blackpanther9229 committed:
adjust add function, create readTask function

blackpanther9229 committed:
Nah.java : Adjust add() method, create readTask() method

blackpanther9229 committed:
create class Task

blackpanther9229 committed:
Create class Task

blackpanther9229 committed:
fix access modifiers in Task

blackpanther9229 committed:
Task.java : adjust access modifiers

blackpanther9229 committed:
fix acces modifiers in Nah class, adjust readTask function, create mark and unmark function and adjust the main function

blackpanther9229 committed:
Nah.java : Adjust readTask() method, create mark() and unMark() method

blackpanther9229 committed:
Create ToDos class

blackpanther9229 committed:
Create ToDos class

blackpanther9229 committed:
Create Deadlines and Events class

blackpanther9229 committed:
Create Deadlines and Events class

blackpanther9229 committed:
Adjust Deadlines and Events class

blackpanther9229 committed:
adjust Deadlines and Events class

blackpanther9229 committed:
Adjust add() method in Nah.java and adjust Events.java

blackpanther9229 committed:
adjust Events class, adjust add function in Nah and adjust main to take input correctly

blackpanther9229 committed:
Adjust runtest.bat and runtest.sh

blackpanther9229 committed:
Adjust runtest.bat and runtest.sh

blackpanther9229 committed:
Adjust EXPEXTED.TXT, input.txt and Nah.java

blackpanther9229 committed:
Adjust EXPEXTED.TXT, input.txt and Nah

blackpanther9229 committed:
Nah.java : Some adjustment to remove unnecessary white spaces in the output

blackpanther9229 committed:
adjust Nah to remove unnecessary white spaces in the output, create EXPECTED file

blackpanther9229 committed:
Runtest.bat: Some adjustment to print out "pass" or "fail"

blackpanther9229 committed:
Adjust runtest a bit to print "pass" or "fail"

blackpanther9229 committed:
Create 3 classes inherit Excetion to handle errors

blackpanther9229 committed:
Create 3 classes inherit Exception to handle errors

blackpanther9229 committed:
adjust main function in Nah to take input correctly, update input and output file

blackpanther9229 committed:
Nah.java : Apply Exception to handle errors, update input and output file

blackpanther9229 committed:
Nah.java : Add delete() method

blackpanther9229 committed:
add delete function in Nah class

blackpanther9229 committed:
Adjust input and expected file

blackpanther9229 committed:
Adjust input.txt and EXPECTED.TXT

blackpanther9229 committed:
Apply enum on command

blackpanther9229 committed:
Nah.java : Apply enum on command

blackpanther9229 committed:
Nah.java : Some white space adjustment for better output

blackpanther9229 committed:
Some white space adjustment for better output

blackpanther9229 committed:
Adjust some methods' modifiers and javadoc

blackpanther9229 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Deadlines.java
#	src/main/java/Events.java
#	src/main/java/Nah.java
#	src/main/java/Task.java
#	src/main/java/ToDos.java

blackpanther9229 committed:
Revert "Merge remote-tracking branch 'origin/master'"

This reverts commit 455cdbee040074937865177532f45acf863d5699, reversing
changes made to d53048a5d9487fec71fbab36be39c8faebad2fff.

blackpanther9229 committed:
Reapply "Merge remote-tracking branch 'origin/master'"

This reverts commit 87625e130780069be22791538a73488423a4121c.

blackpanther9229 committed:
Some change after merging

blackpanther9229 committed:
Create Nah.txt as hard disk memory and apply in Nah.java

blackpanther9229 committed:
Apply LocalDateTime to take date and time input

blackpanther9229 committed:
Nah.java : Create dueOn() method

blackpanther9229 committed:
More OPP features

Let's
* create new class Command and inheriting classes
* create UI class to deal with users' interaction
* create Parser class to convert String to meaningfull command
* create  TaskList class to store tasks
* adjust Nah.java

blackpanther9229 committed:
Create package Nah and releveant packages in Nah

blackpanther9229 committed:
Merge branch 'add-gradle-support'

blackpanther9229 committed:
Set up gradle

blackpanther9229 committed:
Create some JUnit tests

Let's create,
* 5 tests for nah.parser.Parser
* 4 tests for nah.storage.Decoder
* 1 test for nah.storage.Storage
* 5 tests for nah.tasklist.TaskList
* 1 test for nah.Nah

blackpanther9229 committed:
Create some JUnit tests

Let's create,
* 5 tests for nah.parser.Parser
* 4 tests for nah.storage.Decoder
* 1 test for nah.storage.Storage
* 5 tests for nah.tasklist.TaskList
* 1 test for nah.Nah

blackpanther9229 committed:
Add new command Find for Nah

Let's create:
* new subclass FindCommand in nah.command.Command
* new method find(String s) in nah.tasklist.TaskList
* 2 new methods in nah.data.Task

blackpanther9229 committed:
Adjustment on Javadoc comments

blackpanther9229 committed:
Adjustment on Javadoc comments

blackpanther9229 committed:
Adjustment to follow Coding Standard

blackpanther9229 committed:
Merge branch 'branch-A-CodingStandard'

blackpanther9229 committed:
Apply GUI to interact with user

blackpanther9229 committed:
Adjustment on GUI management

Let's,
* Extend MainWindow to have a Stage as a root Node
* Customize the appearance of the application

wang-h-z commented on src/main/java/ToDos.java:
```diff
@@ -0,0 +1,9 @@
+public class ToDos extends Task{
```
Small nitpick about capitalisation, I fell Todo fits better than ToDo here.

wang-h-z commented on src/main/java/InvalidTaskNumber.java:
```diff
@@ -0,0 +1,8 @@
+public class InvalidTaskNumber extends Exception{
```
I like how you have handled various possible errors with different exceptions. Keep up the good work!

wang-h-z commented on src/test/java/nah/NahTest.java:
```diff
@@ -1,2 +1,57 @@
-package nah;public class NahTest {
+package nah;
```
I like how extensive and comprehensive your JUnit tests are. Keep up the good work!

wang-h-z commented on src/main/java/Nah.java:
I like the logic you have used in your switch case blocks. Looks good to me.

wang-h-z reviewed (commented):
Overall, I found your code easy to read and was able to understand the logic behind your chatbot. Looks good to me! Keep up the good work!

AbdulrahmanAlRammah commented on src/main/java/nah/Nah.java:
```diff
@@ -0,0 +1,67 @@
+import nah.ui.UI;
+
+/**
+ * Our chatBot
```
Should there be a period at the end of javadocs sentences?

AbdulrahmanAlRammah commented on src/main/java/nah/command/Command.java:
```diff
@@ -0,0 +1,274 @@
+    /**
+     * The CleanCommand class is for cleaning all data.
+     */
+    public static class CleanCommand extends Command {
```
Should the class be indented?

AbdulrahmanAlRammah commented on src/main/java/nah/data/Task.java:
```diff
@@ -0,0 +1,355 @@
+     * Marks the task as done by setting the {@code isDone} flag to {@code true}.
+     */
+
+    public void mark() {
```
Do the guidelines state that setter methods for booleans should be named similar to: setMarked()?

AbdulrahmanAlRammah commented on src/main/java/nah/exceptions/NahException.java:
```diff
@@ -0,0 +1,46 @@
+     */
+    public static class LackDescriptionException extends NahException {
+
+        public LackDescriptionException(String mess) {
```
perhaps lacks would be better than lack?

AbdulrahmanAlRammah commented on src/main/java/nah/parser/Parser.java:
```diff
@@ -0,0 +1,140 @@
+            }
+            return new Command.AddCommand(new Task.ToDos(cmd[1].trim()));
+        }
+        case "deadline": {
```
For the case statements they should have a break statement or //Fallthrough, see coding guidelines. I haven't used them before, but you can double check if your current form is right.

AbdulrahmanAlRammah commented on src/main/java/nah/tasklist/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+ */
+public class TaskList {
+    private LinkedList<Task> tasks;
+    private int taskCount = 0;
```
Constants should be all caps, if it is a variable then only initialize during constructor.

AbdulrahmanAlRammah reviewed (commented):
Nit comments. I like the good it is well packaged too.

blackpanther9229 committed:
New avatar for chatBot and User

blackpanther9229 committed:
New GUI features for chatBot

blackpanther9229 committed:
Add some assertions to ensure the program run properly

There are possible errors which trigger null value or wrong
LocalDateTime value in the program.

Let's
* Add assertion in execute(TaskList, Storage) in Command class
to make sure the TaskList and Storage passing to the method is
not null.
* Add assertion to the constructors in Task class and its
subclasses to make sure the String passing to the constructor is
 not null
* Add assertion to the constructors of Event class to make sure
the start value is before the end value
* Add assertion to the mark(int), delete(int), unMark(int) to make
sure the task in the given position (int) is not null
* And some other small assertions

blackpanther9229 committed:
Improve Code Quality of the project

Let's,
* Add new method checkFirstThreeComponent(String) in nah.storage.decode
to make the code more readable
* Change some javadoc for clarity

blackpanther9229 committed:
Some testings

blackpanther9229 committed:
Apply Streams in listing matching tasks

As it is very complicated to write for loop in readTask(), find()
and dueOn() in nah.tasklist to listing matching tasks, I apply stream

blackpanther9229 committed:
Merge pull request #2 from blackpanther9229/branch-A-Assertions

Add some assertions to ensure the program run properly

blackpanther9229 committed:
Resolve conflict for merging branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/nah/storage/Decoder.java
#	src/main/resources/filePath/Nah.txt

blackpanther9229 committed:
Resolve conflicts when merging branch 'branch-A-Streams'

# Conflicts:
#	src/main/java/nah/tasklist/TaskList.java
#	src/main/resources/filePath/Nah.txt

blackpanther9229 committed:
Some small change in toString() in nah.data.Task

blackpanther9229 committed:
Create .github/workflows/gradle.yml in the root directory

blackpanther9229 committed:
Create .github/workflows/gradle.yml in the root directory

blackpanther9229 committed:
Add a new command for the chatBot : Help

This command help user to know the supported command
type of th chatBot and the format of each command type

We do this by create a new subclass of nah.command.Command
and some other small changes

blackpanther9229 committed:
Some testing

blackpanther9229 committed:
Make Nah chatBot more user friendly by opening a new
Help Window when taking certain input

To do this, let's,
* Create a new class HelpWindow extends Anchorpane
* Create a class HelpWindowResponse to handle input from
HelpWindow

blackpanther9229 committed:
Upgrade Nah to be more user-friendly

To do this, I did some change to GUI and the way Nah
react to special commands

Let's,
* Change the font of Dialogbox to Rotobo and change size
* Edit the color background of HelpWindow
* Add class Command.NahCommand and
Command.ReactCommand to handle special
response

blackpanther9229 committed:
Modify Readme.md for user guidance

blackpanther9229 committed:
update UI.png

blackpanther9229 committed:
Some checkStyle modification

blackpanther9229 committed:
Some modification with nah.storage.Storage

As the filePath of hard disk in nah.storage.Storage doesn't work
when smoke tes the nah.jar, so we have to change it.

Let's,
* Change the hard disk to ./Nah.txt
* Create a new method createNewHardDisk() to create a new
file following the hard disk
* In the load() method, if the Nah.txt file doesn't exist, use
createNewHardDisk() method

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
update README.md

blackpanther9229 committed:
Update README>md

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Make chatBot more user friendly by setting waiting time when closing
window screen

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Update GUI

To make the window screen more user friendly, let's change some
properties to make the window screen automatically resize as follow:
* In the main Window, make the TextField and Send Button stick to the
left-bottom most and right-bottom most, the ScrollPane will take the
rest of the AnchorPane.
* The same to HelpWindow
* Make the DialogBox resize to the AchorPane of MainWindow

blackpanther9229 committed:
Update images in docs

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Update README.md

blackpanther9229 committed:
Update README.md
